src/main/java/seedu/address/model/person/Person.java

```diff
@@ -316,6 +335,35 @@ public boolean isSameCluster(ReadOnlyPerson other) {
     }
 
     @Override
+    public String calcAccruedAmount() {
+        this.lastAccruedDate = new Date(); // update last accrued date
+        double principal = this.getDebt().toNumber();
+        int interest = Integer.parseInt(this.getInterest().toString());
+        double accruedInterest = principal * ((double) interest) / 100;
+        return String.format("%.2f", accruedInterest);
+    }
+
+    @Override
+    public boolean checkUpdateDebt(Date currentDate) {
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(lastAccruedDate);
+        int monthLastAccrued = cal.get(Calendar.MONTH);
+        cal.setTime(currentDate);
+        int currentMonth = cal.get(Calendar.MONTH);
+        // if currentMonth is 1 month ahead of monthLastAccrued
+        if ((currentMonth - monthLastAccrued == 1) || (currentMonth - monthLastAccrued == -11)) {
+            int currentDay = cal.get(Calendar.DAY_OF_MONTH);
+            if (currentDay == 1) { // interest rate only increments debt on 1st day of month
+                return true;
+            } else {
+                return false;
+            }
+        } else {
```

Comment:
dont need else here. Just return false would do.

Reply:
I tried that. But IntelliJ would just show an error saying that some of the if-else conditions do not return the true/false boolean variable. I will just stick with this for now

Reply:
Oh alright sure man